tests/conditional_processing/tests.py

```diff
@@ -24,12 +25,16 @@ def assertFullResponse(self, response, check_last_modified=True, check_etag=True
         if check_last_modified:
             self.assertEqual(response['Last-Modified'], LAST_MODIFIED_STR)
         if check_etag:
-            self.assertEqual(response['ETag'], '"%s"' % ETAG)
+            self.assertEqual(response['ETag'], ETAG)
 
     def assertNotModified(self, response):
         self.assertEqual(response.status_code, 304)
         self.assertEqual(response.content, b'')
 
+    def assertPreconditionFailed(self, response):
```

Comment:
Perhaps this helper could be added in a separate commit so there are less unrelated test changes in this commit (unless it really is related).